src/graphics/renderer.cpp

```diff
@@ -51,13 +52,14 @@ void Renderer::set_drawing_color(const Color& color) {
     }
 }
 
-SDL_Rect Renderer::convert_to_sdl_rect(const BoundingBox2f& box) {
-    const Point top_left = box.get_top_left();
+SDL_Rect Renderer::convert_to_sdl_rect(const BoundingBox2i& box) {
+    const Point2i top_left = box.get_top_left();
+    const int scaling_factor = Grid::ELEMENT_SIZE;
```

Comment:
Renderer should not be dependent on PCG (inclucing Grid class). The same logic applies to Grid - it should not care about its "ELEMENT_SIZE". Please put SCALING_FACTOR as const in the Renderer class and remove the field in the grid class